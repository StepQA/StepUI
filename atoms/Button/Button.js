import React from 'react';
import StyledButtonFactory from './Button.css';

export const Button = ({
    onClick,
    children,
    type,
    loading,
    size,
    shape
}) => {
     const StyledButton = StyledButtonFactory({
         type,
       loading,
       size,
       shape
     })
    return <StyledButton onClick={!loading ? onClick: ""}>{children}</StyledButton>
    
}