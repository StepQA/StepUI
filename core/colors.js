const colors = {
    navy:"#001F3F",
    blue:"#0074D9",
    aqua: "#7FDBFF",
    teal: "#39CCCC",
    olive: "#3D9970",
    green: "#2ECC40",
    lime: "#01FF70",
    yellow: "#FFDC00",
    orange: "#FF851B",
    //TODO find a AAA compliant red combination
    red: "#FF4136",
    maroon: "#85144b",
    fuchsia: "#F012BE",
    purple: "#B10DC9",
    black: "#111111",
    gray: "#AAAAAA",
    silver: "#DDDDDD",
    white: "#FFFFFF"
    
};


export default {
    colors: {
        
    },
    foreground: {
        primary: colors.aqua,
        success: colors.navy,
        danger: colors.white,
        warning: colors.maroon,
        default: colors.navy
    },
    background: {
        primary: colors.navy,
        success: colors.green,
        danger: colors.red,
        warning: colors.yellow,
        default: colors.gray
    }
};
