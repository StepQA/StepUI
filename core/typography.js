export default {
    base: {
        fontFamily: "Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif"
    }
};
