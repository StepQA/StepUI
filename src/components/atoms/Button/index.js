import React from 'react';
import StyledButtonFactory from './Button.css';

const RawButton = ({
    onClick,
    children,
    loading,
    className
}) => <button onClick={!loading ? onClick: ""} className={className}>{children}</button>;

export const Button = StyledButtonFactory(RawButton)