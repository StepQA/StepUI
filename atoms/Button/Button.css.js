import styled from 'styled-components';

import { colors, constants, typography } from '../../core';


export default function({
    type = "default",
    size = "medium",
    shape,
    loading
}) {

    const hasType = constants.types.indexOf(type) !== -1;
    const hasSize = constants.sizes.indexOf(size) !== -1;
    if (!hasType) {
        console.warn(`Invalid button type: ${type}`);
    }
    if (!hasSize) {
        console.warn(`Invalid button size: ${size}`)
    }
    const StyledButton = styled.button `
        border: 0;
        padding: 5px 8px;
        cursor: pointer;
        font-family: ${typography.base.fontFamily};
        color:${colors.foreground[type]};
        background:${colors.background[type]};
        font-size:${typography.sizes[size]};
        ${loading ? "filter:blur(1px); cursor:wait;": "cursor:pointer;"}
        ${shape === "circle" ? "border-radius:50%": ""}
            &:active {
              transition: all 0.2s; 
              filter: brightness(85%);
            }
            &:hover {
                transition: all 0.3s; 
                box-shadow: 0 4px ${colors.foreground[type]};
                top: -2px;
            }
            
        }
        `;
    return StyledButton;
}
