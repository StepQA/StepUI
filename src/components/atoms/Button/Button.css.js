import React from 'react';
import styled from 'styled-components';

import { colors, constants, typography } from '../../core';


export default (RawButton) => {
    return (props) => {
        const {
            type,
            loading,
            size,
            shape
        } = props;
        const hasType = constants.types.indexOf(type) !== -1;
        const hasSize = constants.sizes.indexOf(size) !== -1;
        if (!hasType) {
            console.warn(`Invalid button type: ${type}`);
        }
        if (!hasSize) {
            console.warn(`Invalid button size: ${size}`)
        }
        const active = !loading ? `&:active {
              transition: all 0.2s; 
              filter: brightness(85%);
            }` : "";
        const hover = !loading ? `&:hover {
                transition: all 0.3s; 
                box-shadow: 0 4px ${colors.foreground[type]};
                top: -2px;
            }` : "";
        const StyledButton = styled(RawButton)`
                border: 0;
                padding: 5px 8px;
                cursor: pointer;
                font-family: ${typography.base.fontFamily};
                color:${colors.foreground[type]};
                background:${colors.background[type]};
                font-size:${typography.sizes[size]};
                ${loading ? "filter:blur(1px); cursor:wait;": "cursor:pointer; transition: all 0.3s; "}
                ${shape === "circle" ? "border-radius:50%": ""}
                    ${active}
                    ${hover}
                }
            `;
        return <StyledButton {...props} />;
    };
};
