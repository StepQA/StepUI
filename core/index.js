import colors from './colors';
import constants from './constants';
import typography from './typography';
export {
    colors,
    constants,
    typography
};