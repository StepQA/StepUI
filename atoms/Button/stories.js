import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button } from './Button';
import { constants } from '../../core';

const buttonStories = storiesOf('Buttons', module);


constants.types.forEach(type => buttonStories.add(type.charAt(0).toUpperCase() + type.slice(1),
  () => constants.sizes.map(size =>(
      <div style={{margin:"10px"}}>
            <Button onClick={action(`clicked ${size} ${type} button`)} type={type} size={size}>
                {type.charAt(0).toUpperCase() + type.slice(1)} button
            </Button>
      </div>
      )
)));


constants.types.forEach(type => buttonStories.add(type.charAt(0).toUpperCase() + type.slice(1) + " circular",
  () => constants.sizes.map(size =>(
      <div style={{margin:"10px"}}>
            <Button onClick={action(`clicked ${size} ${type} button`)} type={type} size={size} shape="circle">
                {type.charAt(0).toUpperCase() + type.slice(1)} button
            </Button>
      </div>
      )
)));

constants.types.forEach(type => buttonStories.add(type.charAt(0).toUpperCase() + type.slice(1) + " loading",
  () => constants.sizes.map(size =>(
      <div style={{margin:"10px"}}>
            <Button onClick={action(`clicked ${size} ${type} button`)} type={type} size={size} loading={true}>
                {type.charAt(0).toUpperCase() + type.slice(1)} button
            </Button>
      </div>
      )
)));
