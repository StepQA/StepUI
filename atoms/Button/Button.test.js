import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Button } from './Button';
import { constants } from '../../core';
describe('ensuring regular buttons match snapshot', () => {
  constants.types.forEach(type => {
    constants.sizes.forEach(size => {
      it(`renders the ${size} ${type} button correctly`, () => {
        const tree = renderer
          .create(<Button onClick={Function} type={type} size={size}></Button>)
          .toJSON();
        expect(tree).toMatchSnapshot();

      });
    });
  });
});
describe('ensuring loading buttons match snapshot', () => {

  constants.types.forEach(type => {
    constants.sizes.forEach(size => {
      it(`renders the ${size} ${type} loading button correctly`, () => {
        const tree = renderer
          .create(<Button onClick={Function} loading={true} type={type} size={size}></Button>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
describe("button UI", () => {
  it('renders the correct child elements in buttons', () => {
    //TODO Add actual nodes, rather than text
    const wrapper = mount(
      <Button>Test content</Button>
    );
    const button = wrapper.find("button");
    expect(button.text()).toBe('Test content');
  });
  it('handles button clicks correctly', done => {
    const clickEvent = () => {
      
      done();
    };
    const wrapper = mount(
      <Button onClick={clickEvent}>Test content</Button>
    );
    const button = wrapper.find("button");
    button.simulate("click");
  });
});
