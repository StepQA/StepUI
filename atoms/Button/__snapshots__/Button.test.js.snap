// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensuring loading buttons match snapshot renders the large danger loading button correctly 1`] = `
<button
  className="sc-kEYyzF epnNGE"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the large default loading button correctly 1`] = `
<button
  className="sc-ckVGcZ hFWsAT"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the large primary loading button correctly 1`] = `
<button
  className="sc-kgoBCf kYswgz"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the large success loading button correctly 1`] = `
<button
  className="sc-cMljjf liidPb"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the large warning loading button correctly 1`] = `
<button
  className="sc-eHgmQL ghyAof"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the medium danger loading button correctly 1`] = `
<button
  className="sc-kkGfuU jGcWbf"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the medium default loading button correctly 1`] = `
<button
  className="sc-jKJlTe jcYjQf"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the medium primary loading button correctly 1`] = `
<button
  className="sc-kGXeez gCxVoz"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the medium success loading button correctly 1`] = `
<button
  className="sc-jAaTju focFly"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the medium warning loading button correctly 1`] = `
<button
  className="sc-cvbbAY kIlToQ"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the small danger loading button correctly 1`] = `
<button
  className="sc-iAyFgw dkZyKB"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the small default loading button correctly 1`] = `
<button
  className="sc-eNQAEJ yRLrQ"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the small primary loading button correctly 1`] = `
<button
  className="sc-kpOJdX Erhek"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the small success loading button correctly 1`] = `
<button
  className="sc-jDwBTQ gdDgyX"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the small warning loading button correctly 1`] = `
<button
  className="sc-jWBwVP ljvdll"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the tiny danger loading button correctly 1`] = `
<button
  className="sc-hSdWYo cNOpjl"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the tiny default loading button correctly 1`] = `
<button
  className="sc-hMqMXs gKsxvc"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the tiny primary loading button correctly 1`] = `
<button
  className="sc-dxgOiQ jKHFCg"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the tiny success loading button correctly 1`] = `
<button
  className="sc-gPEVay jRrHOX"
  onClick=""
/>
`;

exports[`ensuring loading buttons match snapshot renders the tiny warning loading button correctly 1`] = `
<button
  className="sc-brqgnP dYnmvc"
  onClick=""
/>
`;

exports[`ensuring regular buttons match snapshot renders the large danger button correctly 1`] = `
<button
  className="sc-htoDjs ilmbyB"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the large default button correctly 1`] = `
<button
  className="sc-ifAKCX gXHimt"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the large primary button correctly 1`] = `
<button
  className="sc-bdVaJa eSpynT"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the large success button correctly 1`] = `
<button
  className="sc-jzJRlG jQKmxs"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the large warning button correctly 1`] = `
<button
  className="sc-gqjmRU kSwMKk"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the medium danger button correctly 1`] = `
<button
  className="sc-dnqmqq fqAHdl"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the medium default button correctly 1`] = `
<button
  className="sc-EHOje bvtvJz"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the medium primary button correctly 1`] = `
<button
  className="sc-bwzfXH hLjXqq"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the medium success button correctly 1`] = `
<button
  className="sc-cSHVUG kdzsqe"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the medium warning button correctly 1`] = `
<button
  className="sc-VigVT ihOnE"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the small danger button correctly 1`] = `
<button
  className="sc-iwsKbI fRZLzk"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the small default button correctly 1`] = `
<button
  className="sc-bZQynM cRdmRb"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the small primary button correctly 1`] = `
<button
  className="sc-htpNat DQIND"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the small success button correctly 1`] = `
<button
  className="sc-kAzzGY jDrEQX"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the small warning button correctly 1`] = `
<button
  className="sc-jTzLTM jcbo"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the tiny danger button correctly 1`] = `
<button
  className="sc-gZMcBi hSPKAV"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the tiny default button correctly 1`] = `
<button
  className="sc-gzVnrw jruqmd"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the tiny primary button correctly 1`] = `
<button
  className="sc-bxivhb ebqhIa"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the tiny success button correctly 1`] = `
<button
  className="sc-chPdSV gSWEDB"
  onClick={[Function]}
/>
`;

exports[`ensuring regular buttons match snapshot renders the tiny warning button correctly 1`] = `
<button
  className="sc-fjdhpX jTOykx"
  onClick={[Function]}
/>
`;
